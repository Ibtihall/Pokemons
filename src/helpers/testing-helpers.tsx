import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import { render, RenderOptions } from '@testing-library/react';
import store from '../store';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import { renderHook } from '@testing-library/react-hooks';

export const Wrapper: React.FC = (props:any): JSX.Element => {
  return (
    <Provider store={store}>
      {//@ts-ignore
      <Router>
        {props.children} 
      </Router>}
    </Provider>
  );
};


export const renderWithIntl = (
  ui: ReactElement,
  renderOptions?: RenderOptions,
) => {
  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

export const renderHookWithIntl = <P, R>(hook: (props: P) => R) => {
  return renderHook(hook, { wrapper: Wrapper });
};
