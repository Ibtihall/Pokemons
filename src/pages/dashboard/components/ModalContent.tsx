import React from "react";
import CustomSkeleton from "../../../components/Skeleton";
import { COLORS } from "../../../constants/design";
import { IPokemon, IPokemonType } from "../typings";
import StatsProgressBar from "./progressBar";
import { Button, Content, PokeImage, PokeTypes, Tag, Title, TopContent } from "./styles";

export interface ModalContentProps {
    handleClose?: () => void;
    open?: boolean;
    data: IPokemon;
    isLoading: boolean;
}

export const ModalContent: React.FC<ModalContentProps> = (
    props,
): JSX.Element => {
    const { data, handleClose, isLoading } = props;

    const handlColorByPokeType = (type: string) => {
        let color;
        switch (type) {
            case "fire":
                color = COLORS.YELLOWB;
                break;
            case "grass":
                color = COLORS.NAVY;
                break;
            case "poisson":
                color = COLORS.BLUE_GREY;
                break;
            case "flying":
                color = COLORS.BLUE;
                break;
            case "bug":
                color = COLORS.RED;
                break;
            case "normal":
                color = COLORS.FUNNY;
                break;
            default:
                color = COLORS.GREEN;
        }
        return color;
    }
    return (
        <Content>
            {isLoading 
            ?
                <CustomSkeleton lines={10} />
            :
            <>
                <TopContent>
                    <Title data-testId='poke-title'>{data?.name}</Title>
                    <PokeTypes>
                        {data?.types.map((item: IPokemonType) => (
                            <Tag color={handlColorByPokeType(item.type.name)}>{item.type.name}</Tag>
                        ))}
                    </PokeTypes>
                </TopContent>
                <TopContent>
                    <StatsProgressBar stats={data?.stats} />
                    <PokeImage src={data?.sprites.front_default} alt="" />
                </TopContent>

                <Button onClick={handleClose}>Cancel</Button>
            </>
            }
        </Content>
    );
};

export default ModalContent;