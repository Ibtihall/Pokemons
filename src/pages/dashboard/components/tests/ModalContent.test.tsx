import { cleanup, screen } from '@testing-library/react';
import { renderWithIntl } from '../../../../helpers/testing-helpers';
import ModalContent, { ModalContentProps } from '../ModalContent';
afterEach(cleanup);

describe('ModalContent', () => {
  let props: ModalContentProps;
  beforeEach(() => {
    props = {
    handleClose: jest.fn(),
    data: {
        name: 'fake-name',
        height:6,
        base_experience: 62,
        weight: 45,
        sprites: {
            front_default: 'fake-url',
            back_default: 'fake-url',
            front_shiny: 'fake-url'},
        types: [
            {
                slot:1,
                type: {name:'type1',url:'type2-url'}
            },
            {
                slot:2,
                type: {name:'type2',url:'type1-url'}
            }
        ],
        stats: [
            {
                base_stat: 1,
                effort: 'fake-effort',
                stat: {name:'type1',url:'type2-url'}
             }
        ]
    },
    isLoading: false,
    };
 
  });
  describe('render', () => {
    it('should render correctly', () => {
      const { asFragment } = renderWithIntl(<ModalContent {...props} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('title should display on the page', () => {
    renderWithIntl(<ModalContent {...props} />);
    const title = screen.getByTestId('poke-title');
    expect(title).toBeInTheDocument();
  });
});
