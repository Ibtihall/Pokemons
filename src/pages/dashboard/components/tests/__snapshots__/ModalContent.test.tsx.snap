// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContent render should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ cccewh"
  >
    <div
      class="sc-gsnTZi gWVLbH"
    >
      <div
        class="sc-iBkjds ejjMYP"
        data-testid="poke-title"
      >
        fake-name
      </div>
      <div
        class="sc-jSMfEi irKJqN"
      >
        <div
          class="sc-gKXOVf hqkVlx"
          color="#6FAD37"
        >
          type1
        </div>
        <div
          class="sc-gKXOVf hqkVlx"
          color="#6FAD37"
        >
          type2
        </div>
      </div>
    </div>
    <div
      class="sc-gsnTZi gWVLbH"
    >
      <div
        class="sc-dkzDqf jnkykB"
      >
        <div
          class="sc-hKMtZM kcRLao"
        >
          <div
            class="sc-eCYdqJ jykncb"
          >
            type1
          </div>
          <div
            class="sc-crXcEl dGBSxQ"
          >
            <div
              class="sc-evZas hxkPzg"
            >
              <div
                class="sc-breuTD iRenGZ"
                color="#E2EFD7"
              >
                <div
                  class="sc-hAZoDl iLaklF"
                  color="#6FAD37"
                  style="width: 1%;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <img
        alt=""
        class="sc-papXJ cKlRNL"
        src="fake-url"
      />
    </div>
    <button
      class="sc-ftvSup gMZVcl"
    >
      Cancel
    </button>
  </div>
</DocumentFragment>
`;
